test/e2e/e2e_suite_test.go

```diff
@@ -319,6 +320,15 @@ var _ = SynchronizedAfterSuite(func() {}, func() {
 				LogPath:              filepath.Join(artifactFolder, "clusters", proxy.GetName(), "resources"),
 			})
 		})
+		By("Dump workload resources to artifacts", func() {
+			framework.DumpAllResources(ctx, framework.DumpAllResourcesInput{
+				Lister:               workloadProxy.GetClient(),
+				KubeConfigPath:       workloadProxy.GetKubeconfigPath(),
+				ClusterctlConfigPath: workloadKubeconfig,
+				Namespace:            "kube-system",
+				LogPath:              filepath.Join(artifactFolder, "clusters", workloadProxy.GetName(), "resources"),
+			})
+		})
```

Comment:
`DumpAllResources` tries to dump CAPI resources from a workload cluster (which does not have the CRD's installed or any information of them).

It also does not dump any logs afaik. (it uses the logpath to dump `clusterctl describe`, but that also only makes sense in a management cluster).

Reply:
Thanks @chrischdi. Wanted to reuse that for a quick debug for e2e failure yesterday. As you pointed, this won't help for workload cluster. Let me fix it and reopen this PR later.